numba_dpex/experimental/launcher.py

```diff
@@ -229,70 +125,85 @@ def get_queue_ref_val(
         for arg_num, argty in enumerate(kernel_argtys):
             if isinstance(argty, DpnpNdArray):
                 llvm_val = kernel_args[arg_num]
-                datamodel = (
-                    self._cpu_codegen_targetctx.data_model_manager.lookup(argty)
-                )
+                datamodel = self.context.data_model_manager.lookup(argty)
                 sycl_queue_attr_pos = datamodel.get_field_position("sycl_queue")
-                ptr_to_queue_ref = self._builder.extract_value(
+                ptr_to_queue_ref = self.builder.extract_value(
                     llvm_val, sycl_queue_attr_pos
                 )
                 break
 
         return ptr_to_queue_ref
 
-    def get_kernel(self, kernel_module, kbref):
+    def get_kernel(self, qref, kernel_module: _KernelModule):
         """Returns the pointer to the sycl::kernel object in a passed in
         sycl::kernel_bundle wrapper object.
         """
-        kernel_name = self._cpu_codegen_targetctx.insert_const_string(
-            self._builder.module, kernel_module.kernel_name
+        # Inserts a global constant byte string in the current LLVM module to
+        # store the passed in SPIR-V binary blob.
+        kernel_bc_byte_str = self.context.insert_const_bytes(
+            self.builder.module,
+            bytes=kernel_module.kernel_bitcode,
         )
-        return sycl.dpctl_kernel_bundle_get_kernel(
-            self._builder, kbref, kernel_name
+
+        # Send it to dpctl, so it create this function on device and returns
```

Comment:
How about:

Create a sycl::kernel_bundle object and return it as an opaque pointer using dpctl's libsyclinterface.